cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -2207,6 +2213,12 @@ const char* right_utf8_int32(gdv_int64 context, const char* text, gdv_int32 text
     return "";
   }
 
+  //case where right('abcdef', -6) -> "" and right('abcdef', -7) -> ""
+  if(number < 0 && abs(number) >= text_len) {
```

Comment:
This is not handling mutibyte unicode characters

Reply:
I tested it for left('abc¥', -4) and it is returning correct result, Would it better if I put this condition in the  for loop below
if (number < 0 && char_count <= abs(number)) {
      *out_len = 0;
      return "";
    }

Reply:
text_len for strings with multibyte characters will be number of bytes rather than number of characters.. while `number` refers the character count

Reply:
Ok, I assumed text_len is no of characters.